app/src/main/java/com/example/stromtracker/MainActivity.kt

```diff
@@ -42,46 +36,64 @@ class MainActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
         val toolbar: Toolbar = findViewById(R.id.toolbar)
-        var geraeteViewModel: GeraeteViewModel =  ViewModelProviders.of(this).get(GeraeteViewModel::class.java)
-        val haushaltItems:ArrayList<Haushalt> = ArrayList()
-
-
-
-
-
+        var sharedViewModel: SharedViewModel =
+            ViewModelProviders.of(this).get(SharedViewModel::class.java)
+        val haushaltItems: ArrayList<Haushalt> = ArrayList()
         setSupportActionBar(toolbar)
 
 
-
         val drawerLayout: DrawerLayout = findViewById(R.id.drawer_layout)
         val navView: NavigationView = findViewById(R.id.nav_view)
         val navController = findNavController(R.id.nav_host_fragment)
 
-        geraeteViewModel.getAllHaushalt().observe(
+        sp = navView.menu.findItem(R.id.nav_haushalt).actionView as Spinner
+
+        sharedViewModel.getAllHaushalt().observe(
             this,
             Observer { haushalte ->
                 if (haushalte != null) {
 
                     haushaltItems.clear()
                     haushaltItems.addAll(haushalte)
-                    sp= navView.menu.findItem(R.id.nav_haushalt).actionView as Spinner
-                    val adapter = ArrayAdapter<Haushalt>(this, android.R.layout.simple_spinner_dropdown_item, haushaltItems)
+                    sp = navView.menu.findItem(R.id.nav_haushalt).actionView as Spinner
+                    val adapter = ArrayAdapter<Haushalt>(
+                        this,
+                        android.R.layout.simple_spinner_dropdown_item,
+                        haushaltItems
+                    )
                     sp.adapter = adapter
+                    sp.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+                        override fun onNothingSelected(parent: AdapterView<*>?) {
+                        }
+
+                        override fun onItemSelected(
+                            parent: AdapterView<*>,
+                            v: View,
+                            pos: Int,
+                            id: Long
+                        ) {
+                            sharedViewModel.setHaushalt(haushaltItems[pos])
+                        }
+
+
```

Comment:
Wofür genau wird das jetzt gebraucht?

Reply:
Ich habe es oben eigentlich versucht zu erklären, aber im ViewModel wird über LiveData der derzeitige Haushalt gespeichert.
Hier wird über den Listener das ausgewählte Haushalt in das ViewModle gesetzt

Reply:
Ah ok verstehe